ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/hamburgerMenu.js

```diff
@@ -0,0 +1,577 @@
+            hamburgerMenuWidget: {
+                hamburgerMenu: `.${HamburgerMenu.bemBlock}__dropdown`,
+                hamburgerSubMenu: `.${HamburgerMenu.bemBlock}__submenu`,
+                hamburgerMenuNavLink: `.${HamburgerMenu.bemBlock}__link`,
+                hamburgerMenuCloseNavButton: `.${HamburgerMenu.bemBlock}__button--close`,
+                hamburgerMenuOpenNavButton: `.${HamburgerMenu.bemBlock}__button--open`,
+            },
+            hamburgerMenuMiddleContainer: `.${HamburgerMenu.bemBlock}__middle-container`,
+            hamburgerMenuBottomContainer: {
+                self: `.${HamburgerMenu.bemBlock}__bottom-container`,
+                hamburgerMenuActiveItemTitle: `.${HamburgerMenu.bemBlock}__active-item-title`,
+                hamburgerMenuNextNavButton: `.${HamburgerMenu.bemBlock}__button--next`,
+                hamburgerMenuPreviousNavButton: `.${HamburgerMenu.bemBlock}__button--prev`,
+                hamburgerMenuNavButtonsContainer: `.${HamburgerMenu.bemBlock}__navigation-container`,
+            },
+        };
+    
+        static cssClasses = {
+            active: `${HamburgerMenu.bemBlock}__item--active`,
+            activeParent: `${HamburgerMenu.bemBlock}__item--activeparent`,
+            hamburgerMenuWidget: {
+                hamburgerMenu: `${HamburgerMenu.bemBlock}__dropdown`,
+                hamburgerSubMenu: `${HamburgerMenu.bemBlock}__submenu`,
+                hamburgerMenuNavLink: `${HamburgerMenu.bemBlock}__link`,
+                hamburgerMenuCloseNavButton: `${HamburgerMenu.bemBlock}__button--close`,
+                hamburgerMenuOpenNavButton: `${HamburgerMenu.bemBlock}__button--open`,
+            },
+        }
+
+        constructor(self) {
+            this.formContainer = self;
+            self._model.subscribe((action) => {
+                const { payload } = action;
+                const { changes, field } = payload;
+                const { items } = field;
+                if (changes && changes.length > 0) {
+                    changes.forEach((change) => {
+                        switch (change.propertyName) {
+                            case "activeChild":
+                                this.handleActiveItem(field);
+                                break;
+                            case "items":
+                                this.hamburgerMenuInstance.handleItemsChange(change, items);
+                                break;
+                            case "visible":
+                                this.updateAttribute(field.id, 'data-cmp-visible', field.visible);
+                                break;
+                            case "enabled":
+                                this.updateAttribute(field.id, 'data-cmp-enabled', field.enabled);
+                                break;
+                        }
+                    });
+                }
+            }, "fieldChanged");
+        }
+
+        updateAttribute(id, attributeName, value) {
```

Comment:
Please only expose the relevant public methods, other methods should be marked as private